cdflow_commands/ecs_monitor.py

```diff
@@ -88,30 +91,28 @@ def __next__(self):
 
         deployments = self._get_deployments(ecs_service_data)
         primary_deployment = self._get_primary_deployment(deployments)
-        release_image = self._get_release_image(
+
+        self._assert_correct_image_being_deployed(
             primary_deployment['taskDefinition']
         )
 
-        requested_image = '{}:{}'.format(self._component, self._version)
-        if release_image != requested_image:
-            raise ImageDoesNotMatchError(
-                'Requested image {} does not match image '
-                'found in deployment {}'.format(
-                    requested_image, release_image
-                )
-            )
+        # establish whether we're deploying a brand new service, or updating
+        # existing one
+        if self._new_service_deployment is None:
+            self._new_service_deployment = \
+                self._get_previous_running_count(deployments) == 0
```

Comment:
The method might be able to be re-ordered so that it has a fairly obvious flow of: fetch the data, set up variables required for logic, perform logic.